src/frontend/Komponenter/Behandling/VedtakOgBeregning/Skolepenger/VedtakOgBeregningSkolepenger.tsx

```diff
@@ -52,72 +43,55 @@ const VedtakOgBeregningSkolepenger: FC<Props> = ({ behandling, vilkår }) => {
-                                        lagretInnvilgetVedtak={
-                                            vedtakForSkolepenger?._type ===
-                                            IVedtakType.OpphørSkolepenger
-                                                ? vedtakForSkolepenger
-                                                : undefined
-                                        }
-                                        forrigeVedtak={
-                                            vedtakForrigeBehandling &&
-                                            (vedtakForrigeBehandling as unknown as IVedtakForSkolepenger)
-                                        }
-                                    />
-                                );
-                            case EBehandlingResultat.AVSLÅ:
-                                return (
-                                    <AvslåVedtak
-                                        behandling={behandling}
-                                        alleVilkårOppfylt={alleVilkårOppfylt}
-                                        ikkeOppfyltVilkårEksisterer={true}
-                                        lagretVedtak={vedtak}
-                                    />
-                                );
-                            case undefined:
-                            default:
-                                return null;
-                        }
-                    }}
-                </DataViewer>
-            </WrapperMarginTop>
-        </Wrapper>
+            <DataViewer response={{ vedtak, vedtakForrigeBehandling }}>
+                {({ vedtak, vedtakForrigeBehandling }) => {
+                    const vedtakForSkolepenger = vedtak as unknown as IVedtakForSkolepenger;
+                    switch (resultatType) {
+                        case EBehandlingResultat.INNVILGE:
+                        case EBehandlingResultat.OPPHØRT:
+                            // eslint-disable-next-line no-case-declarations
+                            const erOpphør = resultatType === EBehandlingResultat.OPPHØRT;
+                            return (
+                                <VedtaksformSkolepenger
+                                    key={erOpphør ? 'opphør' : 'innvilgelse'}
+                                    behandling={behandling}
+                                    erOpphør={erOpphør}
+                                    lagretInnvilgetVedtak={
+                                        vedtakForSkolepenger?._type ===
+                                        IVedtakType.InnvilgelseSkolepenger
+                                            ? vedtakForSkolepenger
+                                            : undefined
+                                    }
+                                    forrigeVedtak={
+                                        vedtakForrigeBehandling &&
+                                        (vedtakForrigeBehandling as unknown as IVedtakForSkolepenger)
```

Comment:
Ser det ikke er lagt til i denne PR-en, men hva skjer egt her?

Reply:
En grisete måte å blidgjøre TypeScript på, antar jeg 😅 Man sørger for at vedtaket som hentes er av typen `IVedtakForSkolepenger`. Dette kunne nok blitt gjort på en mer forståelig måte...